doc/reference/vm_live_migration_internals.md

```diff
@@ -0,0 +1,104 @@
+
+[`migrateSendLive`](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go#L6666)
+
+This function performs the live migration send process:
+
+1. Connect to the QEMU monitor: The function begins by establishing a connection to the QEMU monitor using `qmp.Connect`.
+1. Define disk names: The function defines names for the root disk (`lxd_root`), the NBD target disk (`lxd_root_nbd`), and the snapshot disk (`lxd_root_snapshot`). These will be used later to manage the root disk and its snapshot during migration.
+1. Check for shared storage: If the migration involves shared storage, the migration process can bypass the need for synchronizing the root disk. The function checks for this condition by verifying if `clusterMoveSourceName` is non-empty and the pool is remote.
+1. Non-shared storage snapshot setup: If shared storage is not used, the function proceeds to set up a temporary snapshot of the root disk.
+  - Migration capabilities such as `auto-converge`, `pause-before-switchover`, and `zero-blocks` are set to optimize the migration process.
+  - The function creates a QCOW2 snapshot file of the root disk, which will store changes to the disk during migration.
+  - The snapshot file is opened for reading and writing, and the file descriptor is passed to QEMU
+  - The snapshot is added as a block device to QEMU, ensuring that it is not visible to the guest OS.
+  - A snapshot of the root disk is taken using `monitor.BlockDevSnapshot`. This ensures that changes to the root disk are isolated during migration.
+1. Revert function: The revert function is used to clean up in case of failure. It ensures that the guest is resumed, and any changes made during snapshot creation are merged back into the root disk if migration fails
+1. Shared storage setup: If shared storage is used, only the `auto-converge` migration capability is set, and no snapshot creation is necessary.
+1. Perform storage transfer: The storage pool is migrated while the VM is still running. The `volSourceArgs.AllowInconsistent` flag is set to true to allow migration while the disk is in use. The migration checks are done by calling `pool.MigrateInstance`.
+1. Notify shared disk pools: For each disk in the VM, the migration process checks if the disk belongs to a shared pool. If so, the disk is prepared for migration by calling MigrateVolume on the source disk.
+1. Setup NBD listener and connection: If shared storage is not used, the function sets up a Unix socket listener for NBD connections. This will handle the actual data transfer for the root disk during migration.
+  - The listener is created, and the source starts listening for incoming NBD connections.
```

Comment:
Here, the bullets are rendering like this:
![image](https://github.com/user-attachments/assets/5c63c101-f623-4375-b594-68730b0d52e1)